src/snowflake/snowpark/mock/_plan.py

```diff
@@ -388,6 +397,32 @@ def handle_function_expression(
     return _MOCK_FUNCTION_IMPLEMENTATION_MAP[func_name](*to_pass_args)
 
 
+def handle_udf_expression(
+    exp: FunctionExpression,
+    input_data: Union[TableEmulator, ColumnEmulator],
+    analyzer: "MockAnalyzer",
+    expr_to_alias: Optional[Dict[str, str]],
+    current_row=None,
+):
+    udf = analyzer.session.udf._registry.get(exp.udf_name)
+
+    if udf is None:
+        raise SnowparkSQLException(
+            f"[Local Testing] udf {exp.udf_name} does not exist."
+        )
+
+    to_pass_args = [
```

Comment:
I've added a test case for that specific scenario and it appears to work. Are there any other edge cases I should be careful about?

Reply:
I'm referring to when inputs of a UDF contains columns that doesn't exist in the `input_data`, for the added test case `test_child_expression`, if we create a udf `double_udf` and evaluate `df.collect(double_udf(col("A") + col("A"))`, this will raise a `KeyError` because `col("A")+col("A")` does not exist in `df`.

Please see here in the stacked PR (support registering from file): https://github.com/snowflakedb/snowpark-python/pull/1325/files#diff-fb4a9902d599442f4f106211e1a05a3de511c8009eb7174d0c96c25aa4c2cf98R415, we need to evaluate the children expressions first.

Reply:
Note: my change doesn't handle when the function input is empty, when udf has no parameters.